### DISCUSSION TITLE
Automatic extensions updates

### DISCUSSION THREAD

**Discussion by ira-gordin-sap** [eclipse-theia/theia#9294] [Q&A] [answered]
Hi,
Does theia have some mechanism of  auto-updates for VS Code extensions installed by user? Some "update.mode" and "extensions.autoUpdate". Something similar as explained here: https://code.visualstudio.com/docs/supporting/faq#_how-do-i-opt-out-of-vs-code-autoupdates
Thanks in advance,
Ira

**Comment by vince-fugnitto** [comment, marked as answer]
@ira-gordin-sap I do not believe that we support auto-updates of plugins yet in the framework.
Can you file an issue track adding support for auto-updating plugins?

**Reply by ira-gordin-sap** [reply]
https://github.com/eclipse-theia/theia/issues/9295

**Reply by ira-gordin-sap** [reply]
Where is settings.json is located in theia?

**Reply by vince-fugnitto** [reply]
> Where is settings.json is located in theia?

@ira-gordin-sap we support preferences at multiple scopes, [similarly to vscode](https://code.visualstudio.com/docs/getstarted/settings#_settings-precedence):
- user - `~/.theia/settings.json`
- workspace - `{workspaceName}/.theia/settings.json`
- folder - `${folderName}/.theia/settings.json`

We also support `.vscode/settings.json` at the workspace and folder scopes if it exists.

**Reply by ira-gordin-sap** [reply]
But "update.mode" and "extensions.autoUpdate" are not allowed there, right?

**Reply by vince-fugnitto** [reply]
> But "update.mode" and "extensions.autoUpdate" are not allowed there, right?

They are not yet implemented, so the values will not have any effect and display a warning.
